projects/ng-components/src/lib/controls/advanced-search/advanced-search.component.html

```diff
@@ -56,8 +69,8 @@
         <ng-container *ngIf="selectedField">
             <!--DO SEARCH-->
             <button type="button"
-                    [ngClass]="{'opacity-50 cursor-not-allowed': fieldValue === undefined && fieldValueDateFrom === undefined && fieldValueDateTo === undefined}"
-                    [disabled]="fieldValue === undefined && fieldValueDateFrom === undefined && fieldValueDateTo === undefined"
+                    [ngClass]="{'opacity-50 cursor-not-allowed': fieldValueDateFrom === undefined && fieldValueDateTo === undefined && (fieldValue === undefined || (operatorsDisabled || selectedOperator === undefined)) && (fieldValue === undefined || !operatorsDisabled)}"
+                    [disabled]="fieldValueDateFrom === undefined && fieldValueDateTo === undefined && (fieldValue === undefined || (operatorsDisabled || selectedOperator === undefined)) && (fieldValue === undefined || !operatorsDisabled)"
```

Comment:
Αυτό εδώ είναι εξαιρετικά πολύπλοκο για να παραμείνει έτσι.
Από ότι καταλαβαίνω:
1) fieldValueDateFrom === undefined && fieldValueDateTo === undefined αυτό έχει να κάνει με την περίπτωση που έχω επιλέξει φίλτρο ημερομηνίας αλλά δεν έχω βάλει τιμή, ας το κάνουμε encapsulate τουλάχιστον σε μια μέθοδο
2) το δεύτερο κομμάτι σπάει σε 2 περιπτώσεις(?): 
operatorsEnabled -> fieldValue και selectedOperator  δεν πρέπει να είναι κανένα από τα 2 undefined
operatorsDisabled -> fieldValue δεν πρέπει να είναι undefined
Μήπως με ένα ternary να απλοποιηθεί η κατάσταση;

Σε κάθε περίπτωση είμαστε φαουλ εδώ (https://medium.com/showpad-engineering/why-you-should-never-use-function-calls-in-angular-template-expressions-e1a50f9c0496), αλλά ας κάνουμε κάπως readable αυτόν τον έλεγχο